delegation_tool.py

```diff
@@ -0,0 +1,387 @@
+                else:
+                    LOGGER.log(LogLevel.INFO, f"Okay, skipping update of existing page '{page_name}'")
+                    return False
+        return True
+    
+
+    def import_full_survey(self, path):
+        '''
+        Places an entire json full of survey data (found at path) in the survey
+        '''
+        LOGGER.log(LogLevel.WARN, "This overwrites all current survey data. Are you sure you want to continue? (y/n)")
+        resp = input("")
+        if resp.strip() in ['y', 'Y']:
+            f = open(path, 'r')
+            self.cur_import_file = path
+            data = json.load(f)
+            self.survey = data
+            self.changes_summary.append(f"Survey data overwritten using '{path}'")
+            LOGGER.log(LogLevel.INFO, f"Imported survey data from '{path}'")
+            f.close()
+        else:
+            LOGGER.log(LogLevel.INFO, "Not importing survey data")
+
+
+if __name__ == '__main__':
+    LOGGER.log(LogLevel.CRITICAL_INFO, 'Welcome to the Delegation Survey Tool')
+    LOGGER.log(LogLevel.INFO, "To get started, enter the version number you'd like to work on:")
+    resp = float(input(""))
+    tool = DelegationTool(resp)
+    while resp != 'q':
+        try:
+            print()
+            LOGGER.log(LogLevel.CRITICAL_INFO, "What would you like to do?\033[37m \n\t\n\tImport Survey from JSON (is)\n\tImport Page from JSON (ip)\n\tDelete Page (dp)\n\tView Current Survey as JSON (v)\n\tSave Changes (s)\n\tQuit (q)")
+            # future options: \n\tAdd Page (ap)\n\tAdd Element to Page (ae)\n\tDelete Element From Page (de)
+            resp = input("").strip().lower()
+            if resp == 'is':
+                if tool.cur_import_file != '':
+                    path = input(f"Enter the path to the import file (enter for {tool.cur_import_file}): ")
+                else:
+                    path = input("Enter the path to the import file: ")
+                tool.import_full_survey(path if path != '' else tool.cur_import_file)
+            if resp == 'ip':
+                if tool.cur_import_file != '':
+                    path = input(f"Enter the path to the import file (enter for {tool.cur_import_file}): ")
+                else:
+                    path = input("Enter the path to the import file: ")
+                page_name = input("Enter the name of the page to import: ")
+                ind = input("At what index should this page be inserted? (enter to append) ")
+                tool.import_page_from_json(path if path != '' else tool.cur_import_file, page_name, int(ind) if ind != '' else None) 
+            if resp == 'dp':
+                page_name = input("Enter the name of the page to delete (only the first page found with this name will be deleted): ")
+                tool.delete_page(page_name)
+            if resp == 's':
+                tool.push_changes()
+            if resp == 'v':
+                print(json.dumps(tool.survey, indent=4))
```

Comment:
This isn't very useful as-is, because the images just spew text on the screen.  As an enhancement request (either for now or down the line), before printing, it would be nice to remove `imgUrl` or similar keys from the result of `json.dumps` or replace their values with a placeholder, like `Image data omitted.`

Reply:
Oh that's interesting. I don't actually store the image data in mongo like that (it is pulled out and there's just a placeholder there) so yes, I should not be printing it. Thanks for catching that!

Reply:
I had to pipe this out to a file in order to see it all. In future iterations, I may look into automatically printing with some sort of more/less command, and I plan to add a print_to_json feature as well, that way we can see what's in the database easily.